.changeset/poor-guests-heal.md

```diff
@@ -0,0 +1,5 @@
+---
+'@wagmi/core': patch
```

Comment:
@jxom thinking this should probably be `minor`, but curious what you think about as `patch`?

Reply:
It is technically a breaking change, but I wonder how breaking it is. 

Folks could be doing `isSuccess && data.status === 0`, so maybe we should play it safe and mark it as a `minor`?